### ISSUE TITLE
Intensity Shuttle Drop Frame

### ISSUE THREAD

**Issue by ChrisNicols** [amiaopensource/vrecord#301]
Hey there. I've been trying to get my Black Magic Intensity Shuttle up and running with vrecord, but I'm getting a lot of drop frame issues. Morgan Morel from BAVC was nice enough to speak with me a few times to run through my set up and process, but we're both stumped and now he suggested I post here. I know that Blackmagic says drop frame is usually a drive speed issue, but Blackmagic's own speed test app says both of my drives should be capable of writing the low range of 4K video.

**Computer-**
Late 2015 iMac, running High Sierra
3.2 GHz Intel Core i5
16GB 1867 MHz DDR3
AMD Radeon R9 M390 2048 MB
Blackmagic Intensity Shuttle (Thunderbolt)
Internal Fusion Drive (alternately sometimes writing out to a Samsung T5 external SSD connected with USB 3.0)

**The Issue-**
I'm getting either 2 or 4 frames dropped within the first few seconds of about 30-40% of all of my captures, regardless of source or destination. I have used a signal generator, a umatic tape run through a TBC and a betacam tape, all with roughly the same results. Here is an excerpt from one such capture's decklink_input log:

[decklink @ 0x7fe0cb800a00] Frame received (#2) - No input signal detected - Frames dropped 1
[decklink @ 0x7fe0cb800a00] Frame received (#3) - Input returned - Frames dropped 2

Although I can't get as detailed a report from Media Express, it also exhibits drop frame errors, so its not a problem that just vrecord is having. I have run 120 short captures of about 30 seconds each and the vast majority of times, it is either the 2nd and 3rd frame or the 3rd and 4th frame that drop. I have also conducted about two dozen 60 minute captures with the same results, never reporting any errors past the first few seconds. Even when capturing at an 8-bit depth and in ProRes, I experience the same problem with the same level of frequency.

If anyone has any idea what might be happening, or if there are certain reports you would need me to generate, please let me know!

**Comment by ChrisNicols** [comment]
Its either 10.9.11 or 10.9.12 since it just updated last Monday. I'll have to double check next week. I have not tried rolling back to earlier versions of the driver, yet.

**Comment by retokromer** [comment]
I’m trying to reproduce, but failed until now.

**Comment by ChrisNicols** [comment]
Hmmm... So I haven't tested this on another computer. Still waiting on procurement for extra cables.

**Comment by ChrisNicols** [comment]
Okay, I've gotten back in and tried running through earlier drivers with the same results. Anything older than 10.9.3 and vrecord couldn't locate the deck and asked for a newer version of the Blackmagic driver. Once the extra cables come in, I will try this on another computer.

**Comment by dericed** [comment]
Ya, vrecord currently uses the ffmpeg decklink device and that only supports the newer decklink drivers. Have you tried the same attempt in Media Express? Media Express won't provide the same verbosity but could show the same result (missing frames). Pinging @dheitmueller for any other debugging ideas.

**Comment by ChrisNicols** [comment]
I'm running a test with Media Express using the 10.8.6 driver, since its the most recent non 10.9.x driver. Of course, the great thing about using vrecord for testing was that I could tell if there would be dropped frames almost instantly. Media Express can go on for over 60 minutes without reporting a drop frame issue.

**Comment by dericed** [comment]
Here's a example of the log: https://gist.github.com/dericed/84a99d942c5302a48532e7355f3df19d#file-vrecord-log-L47-L53.

Notable excerpt:
```
Opening an input file: Intensity Shuttle Thunderbolt.
[decklink @ 0x7fc926800a00] Trying to find mode for frame size 0x0, frame timing 0/0, field order 0, direction 0, mode number 0, format code ntsc
[decklink @ 0x7fc926800a00] Found Decklink mode 720 x 486 with rate 29.97(i)
[decklink @ 0x7fc926800a00] Using 8 input audio channels
[decklink @ 0x7fc926800a00] Frame received (#1) - No input signal detected - Frames dropped 1
[decklink @ 0x7fc926800a00] Frame received (#3) - Input returned - Frames dropped 2
[decklink @ 0x7fc926800a00] Probe buffer size limit of 5000000 bytes reached
```

**Comment by dericed** [comment]
@ChrisNicols anything else to resolve here? okay to close?

**Comment by ChrisNicols** [comment]
I haven’t tested it since. My equipment is off to be repaired, so I might revisit it once it comes back and see if anything’s changes.

**Comment by retokromer** [comment]
Thank you, @ChrisNicols! There was quite some refactoring since you opened the issue… Please report back once you could test the current release (or the head).

**Comment by dericed** [comment]
Hi @ChrisNicols, I'm going to close this issue as it seems to be a duplicate of https://github.com/amiaopensource/vrecord/issues/474. Could you test if syncing your preferences between Desktop Video Setup and vrecord addresses this, if not or you don't find https://github.com/amiaopensource/vrecord/issues/474 related then feel welcome to re-open this.
